crates/contracts/solidity/Trader.sol

```diff
@@ -62,29 +63,36 @@ contract Trader {
     receive() external payable {}
 
     /// @dev Executes needed actions on behalf of the trader to make the trade possible.
-    ///      (e.g. wrapping ETH and setting approvals)
+    ///      (e.g. wrapping ETH, setting approvals, and funding the account)
     /// @param settlementContract - pass in settlement contract because it does not have
     /// a stable address in tests.
     /// @param sellToken - token being sold by the trade
     /// @param sellAmount - expected amount to be sold according to the quote
     /// @param nativeToken - ERC20 version of the chain's native token
+    /// @param spardose - piggy bank for requesting additional funds
     function prepareSwap(
         ISettlement settlementContract,
         address sellToken,
         uint256 sellAmount,
-        address nativeToken
+        address nativeToken,
+        address spardose
     ) external {
         require(!alreadyCalled(), "prepareSwap can only be called once");
 
         if (sellToken == nativeToken) {
             uint256 availableNativeToken = IERC20(sellToken).balanceOf(address(this));
             if (availableNativeToken < sellAmount) {
                 uint256 amountToWrap = sellAmount - availableNativeToken;
-                require(address(this).balance >= amountToWrap, "not enough ETH to wrap");
-                // Simulate wrapping the missing `ETH` so the user doesn't have to spend gas
-                // on that just to get a quote. If they are happy with the quote and want to
-                // create an order they will actually have to do the wrapping, though.
-                INativeERC20(nativeToken).deposit{value: amountToWrap}();
+                // If the user has sufficient balance, simulate the wrapping the missing
```

Comment:
Given that faking `WETH` balances is very reliable I think we should eventually get rid of this special case entirely.
That would reduce the needed solidity code and, more importantly, the number of arguments we have to pass into the call.
Maybe a follow up tasks once this is actually configured in the infra repo and ran successfully for a while.

Reply:
> Given that faking WETH balances is very reliable I think we should eventually get rid of this special case entirely.

The idea behind the special case is that quote verification works even with token balance overrides disabled when the account has sufficient ETH. Personally, I think the path makes sense (the account has the required funds, they are just not ready to swap, like an allowance not being set).